### PR TITLE
Chapter7

### PR DESCRIPTION
**Pull request by xtrxllx**
ready

### CODE REVIEW

**Comment by xtrxllx** [comment]
ready

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/users.scss]: remove empty file

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: please handle when @user is nil

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (ngocvt-0484) [inline app/helpers/users_helper.rb]: remove empty file

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 Rails.application.routes.draw do
-  root "chapter3#home"
-  get "chapter3/help"
+  root "static_pages#home"
 
+  get "/help", to: "static_pages#help"
+  get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
+  get "users/show"
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it because you have resources :users

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
+  get "users/new"
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it because you have resources :users

**Comment by xtrxllx** [comment]
ready

File: app/assets/stylesheets/sessions.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/sessions.scss]: remove file that never used

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: remove blank line

File: app/helpers/sessions_helper.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  
+    def logged_in?
+      current_user.present?
+    end
+  
+    def log_out
+      session.delete :user_id
+      @current_user = nil
+    end
+  end
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]: indenting only 2 spaces and add endline

File: app/models/user.rb [hunk 9]
```diff
@@ -1,9 +1,32 @@
 class User < ApplicationRecord
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline app/models/user.rb]: resolve confict, fplease

File: app/views/sessions/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+        <%= form_for :session, url: login_path do |f| %>
+          <%= f.label :email %>
+          <%= f.email_field :email, class: "form-control" %>
+          <%= f.label :password %>
+          <%= f.password_field :password, class: "form-control" %>
+          <%= f.submit "Log in", class: "btn btn-primary" %>
+        <% end %>
+        <p><%= t "new_user?" %><%= link_to t("sign_up_now!"), signup_path %></p>
+      </div>
+   </div>
```
Comment (saobv-2223) [inline app/views/sessions/new.html.erb]: those line you dont need to indent them

File: app/views/users/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      <%= f.label :email %>
+      <%= f.email_field :email %>
+      <%= f.label :password %>
+      <%= f.password_field :password %>
+      <%= f.label :password_confirmation, "confirmation" %>
+      <%= f.password_field :password_confirmation %>
+      <%= f.submit "create my account!", class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
```
Comment (saobv-2223) [inline app/views/users/new.html.erb]: add endline

**Comment by xtrxllx** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**